pkg/secrets/pushtofile/secret_group.go

```diff
@@ -108,21 +109,22 @@ func (sg *SecretGroup) absoluteFilePath(secretsBasePath string) (string, error)
 		)
 	}
 
-	filePathIsDir := strings.HasSuffix(filePath, "/")
-
-	// fileTemplate requires filePath to point to a file (not a directory)
-	if filePathIsDir && len(fileTemplate) > 0 {
-		return "", fmt.Errorf(
-			"provided filepath %q for secret group %q must specify a path to a file, without a trailing %q",
-			filePath, groupName, "/",
-		)
-	}
-	// Without the restrictions of fileTemplate, the filename defaults to "{groupName}.{fileFormat}"
-	if filePathIsDir && len(fileTemplate) == 0 {
-		filePath = path.Join(
-			filePath,
-			fmt.Sprintf("%s.%s", groupName, fileFormat),
-		)
+	pathContainsFilename := !strings.HasSuffix(filePath, "/") && len(filePath) > 0
+
+	if !pathContainsFilename {
+		if len(fileTemplate) > 0 {
```

Comment:
I think avoiding nested conditions aids in readability. There are some thoughts around this (re: line of sight) in our style guide. https://github.com/cyberark/community/blob/main/Conjur/conventions/golang-style-guide.md#line-of-sight-coding-best-practice

Reply:
@doodlesbykumbi,  I'm thinking it would be acceptable if we simply eliminate the `} else {` indentation below. (It's not needed since we're returning in the conditional above it.)

Is this what you meant? Or are you suggesting something of the form:
```
    if !pathContainsFilename && len(fileTemplate) > 0 {
        return someError
    }
    if !pathContainsFilename {
        doTheJoinThing
    }
```